### PR TITLE
fix workaround for revision validation bug

### PR DESCRIPTION
**Pull request by Monkeyanator**
Previous documented workaround was flawed for users on in-cluster operator (per [this comment](https://github.com/istio/istio/pull/30286#issuecomment-786595621)). This removes the incorrect workaround and gives a code snippet for users to help with the other workaround

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/setup/upgrade/canary/index.md [hunk 1]
```diff
@@ -57,11 +57,14 @@ istio-sidecar-injector-canary   1          3m18s
 {{< /text >}}
 
 {{< warning >}}
-Due to [a bug](https://github.com/istio/istio/issues/28880) in the creation of the `ValidatingWebhookConfiguration` during install, initial installations of Istio __must not__ specify a revision. As a temporary workaround, for Istio resource validation to continue working after removing the non-revisioned Istio installation, the `ValidatingWebhookConfiguration` must be manually pointed to the revision that should handle validation.
+Due to [a bug](https://github.com/istio/istio/issues/28880) in the creation of the `ValidatingWebhookConfiguration` during install, initial installations of Istio __must not__ specify a revision. As a temporary workaround, for Istio resource validation to continue working after removing all revisioned Istio installations, the `istiod` service must be manually pointed to the revision that should handle validation.
```
Comment (smawson) [inline content/en/docs/setup/upgrade/canary/index.md:60]: I think the text is wrong here, the workaround is for after you remove the non-revisioned install, not when you remove revisioned installs.

Comment (Monkeyanator) [reply content/en/docs/setup/upgrade/canary/index.md]: Good catch thanks, the phrasing was indeed backwards. Fixed now

File: content/en/docs/setup/upgrade/canary/index.md [hunk 2]
```diff
@@ -59,9 +59,12 @@ istio-sidecar-injector-canary   1          3m18s
 {{< warning >}}
 Due to [a bug](https://github.com/istio/istio/issues/28880) in the creation of the `ValidatingWebhookConfiguration` during install, initial installations of Istio __must not__ specify a revision. As a temporary workaround, for Istio resource validation to continue working after removing the non-revisioned Istio installation, the `ValidatingWebhookConfiguration` must be manually pointed to the revision that should handle validation.
```
Comment (smawson) [inline content/en/docs/setup/upgrade/canary/index.md:60]: I liked the other part of the change here though, telling the user they need to create an istiod service, I thought that was the point of the bug? It isn't enough just to change ValidatingWebhookConfiguration, you need to create a revision-less istiod service and point it the right place?

Comment (smawson) [reply content/en/docs/setup/upgrade/canary/index.md:60]: Suggested text:

As a temporary workaround, for Istio resource validation to continue working after removing the non-revisioned Istio installation, the `istiod` service must be manually pointed to the revision that should handle validation.

Comment (Monkeyanator) [reply content/en/docs/setup/upgrade/canary/index.md]: Yup, makes sense, changed. Wasn't sure if users would see the connection between the `ValidatingWebhook` and the `istiod` service but it's probably better we don't mention "modifying where the webhook points" at all.

**Review by smawson — APPROVED** [review]

**Pull request merged**